### PR TITLE
Update free-programming-books-subjects.md

### PR DESCRIPTION
**Pull request by abhay57**
## What does this PR do?
Add resource(s) 

### Description
Two books have been added through this PR: 
1. Operating system concepts by Avi Silberschatz, Greg Gagne, and Peter Baer Galvin in 'Operating systems' section
2. Combinatorial problems and exercises in 'Mathematics for Computer science' section

### Why is this valuable (or not)?
The operating systems book covers all the necessary topics in great detail. It is an above freshmen level book.
While the mathematics book is a practice book that contains questions and exercises on combinatorial problems.

### How do we know it's really free?
I've provided direct drive link and download link respectively for both the resources.

### For book lists, is it a book? For course lists, is it a course? etc.
It is a book

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -519,6 +519,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 ### Mathematics For Computer Science
 
 * [A Mathematical Theory of Communication](https://archive.org/details/bstj27-4-623) - Claude E.Shannon
+* [Combinatorial Problems And Exercises](https://archive.org/details/in.ernet.dli.2015.141538/page/n25/mode/2up) - Lovasz,l
```
Comment (LuigiImVector) [inline books/free-programming-books-subjects.md:522]:
```suggestion
* [Combinatorial Problems And Exercises](https://archive.org/details/in.ernet.dli.2015.141538/page/n25/mode/2up) - L. Lovasz *(:card_file_box: archived)*
```

File: books/free-programming-books-subjects.md [hunk 2]
```diff
@@ -649,6 +650,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [How to write a simple operating system in assembly language](http://mikeos.sourceforge.net/write-your-own-os.html) - Mike Saunders (HTML)
 * [Linux From Scratch](https://www.linuxfromscratch.org/lfs/view/stable/) - Gerard Beekmans, Bruce Dubbs, Ken Moffat, Pierre Labastie, et al. (HTML, [PDF, downloads...](https://www.linuxfromscratch.org/lfs/downloads/stable/))
 * [Making Servers Work: A Practical Guide to Linux System Administration](https://www.digitalocean.com/community/books/sysadmin-ebook-making-servers-work) - Jamon Camisso (PDF, EPUB)
+* [Operating System concepts](https://drive.uqu.edu.sa/_/mskhayat/files/MySubjects/2017SS%20Operating%20Systems/Abraham%20Silberschatz-Operating%20System%20Concepts%20(9th,2012_12).pdf) - Avi Silberschatz, Greg Gagne, and Peter Baer Galvin (PDF)
```
Comment (LuigiImVector) [inline books/free-programming-books-subjects.md:653]: is it really free? no illegal resources

Comment (davorpa) [reply books/free-programming-books-subjects.md:653]: Remove illegal content

Publisher website: https://www.wiley.com/en-us/Operating+System+Concepts%2C+10th+Edition-p-9781119320913

```suggestion
```

**Review by davorpa — APPROVED** [review]

File: books/free-programming-books-subjects.md [hunk 3]
```diff
@@ -519,6 +519,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 ### Mathematics For Computer Science
 
 * [A Mathematical Theory of Communication](https://archive.org/details/bstj27-4-623) - Claude E.Shannon
+* [Combinatorial Problems And Exercises](https://archive.org/details/in.ernet.dli.2015.141538/page/n25/mode/2up) - L. Lovasz *(:card_file_box: archived)*
```
Comment (eshellman) [inline books/free-programming-books-subjects.md:522]: why are you linking to page 29? Please find a link for the book, not a page in a book

Comment (davorpa) [reply books/free-programming-books-subjects.md:522]: Using most authoritative url linking to bookcover :wink:

```suggestion
* [Combinatorial Problems And Exercises (1979)](https://archive.org/details/in.ernet.dli.2015.141538/) - L. Lovasz *(:card_file_box: archived)*
```

**Pull request merged**